backend/src/main/java/saci/application/LevelController.java

```diff
@@ -1,30 +1,51 @@
 package saci.application;
 
 import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.ArraySchema;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
 import io.swagger.v3.oas.annotations.responses.ApiResponse;
 import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import java.util.List;
 import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.DeleteMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+import saci.domain.model.Level;
 import saci.domain.service.LevelService;
 import saci.domain.service.exceptions.NotFoundException;
-import io.swagger.v3.oas.annotations.media.Content;
-import io.swagger.v3.oas.annotations.media.Schema;
-import org.springframework.http.HttpStatus;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.ResponseStatus;
-import saci.domain.model.Level;
 
 @RestController
 @RequestMapping("/api/levels")
 @RequiredArgsConstructor
 public class LevelController {
 
-    private final LevelService LevelService;
+    private final LevelService levelService;
+
+    @Operation(summary = "Get all of the Levels")
+    @ApiResponses(
+            value = {
+                @ApiResponse(
+                        responseCode = "200",
+                        description = "List of all Levels",
+                        content = {
+                            @Content(
+                                    mediaType = "application/json",
+                                    array =
+                                            @ArraySchema(
+                                                    schema = @Schema(implementation = Level.class)))
+                        })
+            })
+    @GetMapping
+    public ResponseEntity<List<Level>> getLevels(@RequestParam(required = false) Long role_id) {
+        List<Level> levels;
+        if (role_id != null) {
+            levels = levelService.getLevelsByRoleId(role_id);
+        } else {
+            levels = levelService.getLevels();
```

Comment:
In any of the frontends are you calling this endpoint without roleId? 
If not you don't need this logic (same in the service).